### DISCUSSION TITLE
OC Modules Areas / SubAreas

### DISCUSSION THREAD

**Discussion by Skrypt** [OrchardCMS/OrchardCore#4018] [Q&A]
Just a quick reminder that we should support Areas in OrchardCore modules. Right now each module is an Area. Though when you create an Area folder in an OC module right now this is not supported and quite buggy.

Open discussion on the topic. Should it be supported by folder convention like normally in MVC or should we just keep one module = one area.

/cc @jtkech @sebastienros

**Comment by themattkenney** [comment]
For us we added areas to one of our custom modules. On our local machines they work fine, but when you deploy to Azure they fail. We added areas to make code separation cleaner in this module as we added more views. Without knowledge that this didn’t work it was a very confusing bug to find Azure and I can see other devs struggling once they deploy to an Azure web app.

**Comment by themattkenney** [comment]
The only way we finally figured out what was causing the issue is when @Skrypt let us know this wasn’t supported as the errors on Azure were very generic and hard to track down.

**Comment by jtkech** [comment]
Because a module is already an area, we would need to implement a subarea system with the related attributes, view location expanders and so on. Hmm, maybe not so easy but i'm open to work on it when i will have time and if it becomes a demand and get included in the roadmap.
